Sources/SwiftSyntax/SyntaxNodes.swift.gyb.template

```diff
@@ -100,11 +100,9 @@ public struct ${node.name}: ${base_type}Protocol, SyntaxHashable {
 %
   public let _syntaxNode: Syntax
 
-  /// Converts the given `Syntax` node to a `${node.name}` if possible. Returns
-  /// `nil` if the conversion is not possible.
-  public init?(_ syntax: Syntax) {
-    guard syntax.raw.kind == .${node.swift_syntax_kind} else { return nil }
-    self._syntaxNode = syntax
+  public init?<S: SyntaxProtocol>(_ syntaxNode: S) {
+    guard syntaxNode.raw.kind == .${node.swift_syntax_kind} else { return nil }
+    self._syntaxNode = Syntax(syntaxNode)
```

Comment:
Personally, `self._syntaxNode = syntaxNode._syntaxNode` looks more clear what this is doing.